services/api/Tweek.ApiService.NetCore/Startup.cs

```diff
@@ -113,6 +119,75 @@ public void ConfigureServices(IServiceCollection services)
+            InputValidationContextDriver DecoratorFactory(InputValidationContextDriver.Mode mode, bool reportOnly)
+            {
+                return new InputValidationContextDriver(
+                    decoratedDriver, 
+                    provider.GetService<ILogger<InputValidationContextDriver>>(),
+                    SchemaProvider(provider),
+                    CustomTypeDefinitionProvider(provider),
+                    mode,
+                    reportOnly
+                    );
+            }
+
+            var reportOnlyFromConfig = string.Equals(Configuration["Context:Validation:Error_Policy"], "BYPASS_LOG");
+            var modeRaw =  Configuration["Context:Validation:Mode"];
+            if (string.IsNullOrEmpty(modeRaw)){
+                modeRaw = "Off";
+            }
+            
+            switch(modeRaw){
+                case "Off":
+                    return decoratedDriver;
+                case "Flexible":
+                    return DecoratorFactory(InputValidationContextDriver.Mode.AllowUndefinedProperties, reportOnlyFromConfig);
+                case "Strict":
+                    return DecoratorFactory(InputValidationContextDriver.Mode.Strict, reportOnlyFromConfig);
+                default:
+                    throw new ArgumentException($"Invalid context validation mode ${modeRaw}");
+            }
+        }
+
+        private InputValidationContextDriver.CustomTypeDefinitionProvider CustomTypeDefinitionProvider(IServiceProvider provider)
+        {
+            var tweek = provider.GetService<ITweek>();
+
+            return typeName => {
+                var key = $"@tweek/custom_types/{typeName}";
+                var configuration = provider.GetService<ITweek>().Calculate(new [] { new ConfigurationPath(key)}, new System.Collections.Generic.HashSet<Identity>(),
+                 i => ContextHelpers.EmptyContext);
+
+                ConfigurationValue value;
+
+                var optionValue = configuration.TryGetValue(key, out value) ?
+                        Option<JsonValue>.Some(value.Value) :
+                        Option<JsonValue>.None;
+
+                return optionValue.Map(raw => JsonConvert.DeserializeObject<CustomTypeDefinition> (JsonConvert.SerializeObject(raw, new []{new JsonValueConverter()})));
+                    
+            };   
+        }
+        private InputValidationContextDriver.IdentitySchemaProvider SchemaProvider(IServiceProvider provider)
+        {
+            var tweek = provider.GetService<ITweek>();
+
+            return identityType => {
+                var key = $"@tweek/schema/{identityType}";
+                var configuration = provider.GetService<ITweek>().Calculate(new [] { new ConfigurationPath(key)}, new System.Collections.Generic.HashSet<Identity>(),
+                 i => ContextHelpers.EmptyContext);
+
+                ConfigurationValue value;
+
+                return 
+                    configuration.TryGetValue(key, out value) ?
```

Comment:
There's a LanguageExt extension of "TryGetValue" that return optional.

Reply:
Thanks, fixed.